include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -1620,6 +1622,84 @@ namespace internal
 } // namespace internal
 
 
+
+namespace MGTransferGlobalCoarseningTools
+{
+  template <int dim, int spacedim>
+  std::vector<std::shared_ptr<Triangulation<dim, spacedim>>>
+  create_geometric_coarsening_sequence(
+    const Triangulation<dim, spacedim> &fine_triangulation_in)
+  {
+    std::vector<std::shared_ptr<Triangulation<dim, spacedim>>>
+      coarse_grid_triangulations;
+
+#ifndef DEAL_II_WITH_P4EST
+    (void)fine_triangulation_in;
+    Assert(false, ExcNotImplemented());
+#else
+    const auto fine_triangulation =
+      dynamic_cast<const parallel::distributed::Triangulation<dim, spacedim> *>(
+        &fine_triangulation_in);
+
+    Assert(fine_triangulation, ExcNotImplemented());
+
+    const unsigned int max_level = fine_triangulation->n_global_levels() - 1;
+
+    const auto coarse_mesh_description =
+      GridTools::get_coarse_mesh_description(*fine_triangulation);
+
+    coarse_grid_triangulations.resize(fine_triangulation->n_global_levels());
+
+    coarse_grid_triangulations[max_level].reset(
+      const_cast<Triangulation<dim, spacedim> *>(&fine_triangulation_in),
+      [](auto &) {
+        // empty deleter, since fine_triangulation_in is an external field
+        // and its destructor is called somewhere else
+      });
+
+    // create coarse meshes
+    for (unsigned int l = max_level; l > 0; --l)
```

Comment:
@kronbichler I went with this (and have shifted the indices where they are used). Any objections here.